frontend/public/components/pod.tsx

```diff
@@ -1020,35 +949,103 @@ export const PodsDetailsPage: React.FC<PodDetailsPageProps> = (props) => {
+  const columnLayout = React.useMemo<ColumnLayout>(
     () => ({
-      showNodes,
+      id: columnManagementID,
+      type: t('public~Pod'),
+      columns: columns.map((col) => ({
+        id: col.id,
+        title: col.title,
+        additional: col.additional,
+      })),
+      selectedColumns: new Set(activeColumns.map((col) => col.id)),
     }),
-    [showNodes],
+    [columns, columnManagementID, activeColumns, t],
   );
+  const podStatusFilterOptions = React.useMemo<DataViewFilterOption[]>(
+    () => [
+      { value: 'Running', label: t('public~Running') },
+      { value: 'Pending', label: t('public~Pending') },
+      { value: 'Terminating', label: t('public~Terminating') },
+      { value: 'CrashLoopBackOff', label: t('public~CrashLoopBackOff') },
+      // Use title "Completed" to match what appears in the status column for the pod.
+      // The pod phase is "Succeeded," but the container state is "Completed."
+      { value: 'Succeeded', label: t('public~Completed') },
+      { value: 'Failed', label: t('public~Failed') },
+      { value: 'Unknown', label: t('public~Unknown') },
+    ],
+    [t],
+  );
+  const additionalFilterNodes = React.useMemo<React.ReactNode[]>(
+    () => [
+      <DataViewCheckboxFilter
+        key="status"
+        filterId="status" // is `rowFilter-pod-status`in <FilterToolbar> as a single param, not multiple
+        title={t('public~Status')}
+        placeholder={t('public~Filter by status')}
+        options={podStatusFilterOptions}
+      />,
+    ],
+    [podStatusFilterOptions, t],
+  );
+  const matchesAdditionalFilters = React.useCallback(
+    (resource: PodKind, filters: PodFilters) =>
+      filters.status.length === 0 ||
+      filters.status.includes(
+        String(
+          podStatusFilterOptions.find((option) => option.value === podPhaseFilterReducer(resource))
+            ?.value,
+        ),
+      ),
+    [podStatusFilterOptions],
+  );
+
   return (
-    userSettingsLoaded && (
-      <VirtualizedTable<PodKind, PodRowData>
+    <React.Suspense fallback={<LoadingBox />}>
+      <ResourceDataView<PodKind, PodRowData, PodFilters>
         {...props}
-        aria-label={t('public~Pods')}
-        label={t('public~Pods')}
-        columns={activeColumns}
-        Row={PodTableRow}
-        rowData={rowData}
+        label={PodModel.labelPlural}
+        data={data}
+        loaded={loaded}
+        columns={columns}
+        columnLayout={columnLayout}
+        columnManagementID={columnManagementID}
+        initialFilters={{ ...initialFiltersDefault, status: [] }}
+        additionalFilterNodes={additionalFilterNodes}
+        matchesAdditionalFilters={matchesAdditionalFilters}
+        getDataViewRows={(rowData, tableColumns) =>
+          getDataViewRows(rowData, tableColumns, showNodes, podMetrics)
+        }
+        hideNameLabelFilters={hideNameLabelFilters}
+        hideLabelFilter={hideLabelFilter}
+        hideColumnManagement={hideColumnManagement}
```

Comment:
i believe we don't need to do this here right?
```suggestion
        hideColumnManagement={true}
```

Reply:
We need it as PodList is used elsewhere.